x-pack/plugins/stack_connectors/server/connector_types/crowdstrike/crowdstrike.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ServiceParams, SubActionConnector } from '@kbn/actions-plugin/server';
+import type { AxiosError } from 'axios';
+import { SubActionRequestParams } from '@kbn/actions-plugin/server/sub_action_framework/types';
+import type {
+  CrowdstrikeConfig,
+  CrowdstrikeSecrets,
+  CrowdstrikeGetAgentsResponse,
+  CrowdstrikeGetAgentsParams,
+  CrowdstrikeBaseApiResponse,
+  CrowdstrikeHostActionsParams,
+  CrowdstrikeGetTokenResponse,
+} from '../../../common/crowdstrike/types';
+import {
+  CrowdstrikeGetAgentsResponseSchema,
+  CrowdstrikeHostActionsParamsSchema,
+  CrowdstrikeGetAgentsParamsSchema,
+  CrowdstrikeGetTokenResponseSchema,
+  CrowdstrikeHostActionsResponseSchema,
+} from '../../../common/crowdstrike/schema';
+import { SUB_ACTION } from '../../../common/crowdstrike/constants';
+
+export const API_PATH = 'https://api.crowdstrike.com';
```

Comment:
We chatted briefly offline, but wanted to provide the comment here for visibility.

Its strange to me that we are hardcoding the URL to the API. This (IMO) can have some drawbacks and limitations from a customer's standpoint, including:

- Support on-prem instances of CrowdStrike (if thats a thing)
- Being unable to proxy requests through a corporate gateway (ex. for highly secured environments)
- Supporting air-gaped environments

Reply:
Anything on this comment?

my suggestion: add it to the UI (form) default it to this URL. That gives the user the opportunity to change it if in fact there could be a different URL for the API